main.c

```diff
@@ -556,6 +556,60 @@ exiterr (GError *error)
   exit (EXIT_FAILURE);
 }
 
+/**
+ * apply_excludes:
+ * @pool: initialized pool
+ * @excludes: (out): a map of solvables
+ * @exclude_packages: names of packages to be excluded
+ * @lookasides: set of repos that will not have packages excluded
+ * @modular_pkgs: map of modular packages that exclusion does not apply to
+ *
+ * Apply excludes: this completely hides the package from any processing.
+ * Packages in lookaside are not removed, and neither are modular packages.
+ */
+static void
+apply_excludes(Pool *pool,
+               Map *excludes,
+               GStrv exclude_packages,
+               GHashTable *lookaside_repos,
```

Comment:
I realized that there is better way of having this not passed everywhere around is to repodata_set_void (set some attribute on repo), but it is good for now. Unless you are willing to append PR with a new patch :)